app/models/polling/cert_status_updater.rb

```diff
@@ -0,0 +1,41 @@
+class CertStatusUpdater
+  def initialize(hub_config_api)
+    @hub_config_api = hub_config_api
+  end
+
+  def update_hub_usage_status_for_cert(certificate_to_check)
+    outcomes = entity_ids_for(certificate_to_check).map { |entity_id| cert_is_in_use_for_entity_id?(entity_id, certificate_to_check) }
+
+    if outcomes.all?
+      update_cert_status_for(certificate_to_check)
+    end
```

Comment:
I wonder if we should return true/false here (back to the caller of this method) so it knows whether it's been successful or not...

Reply:
I suppose we could, but I think it'd be better to check the database when deciding whether to retry, rather than trusting a return value from this method as a proxy for the database update.

Reply:
the method will return either `nil` or the event object, right? .. I was just thinking ahead from seeing some of Ola's work, where there's a condition when the polling should stop, so it could just wait for a true value of this method.. but no nil would do too I guess.